src/coreComponents/functions/FunctionBase.hpp

```diff
@@ -137,76 +142,77 @@ class FunctionBase : public dataRepository::Group
    * @param[in] set the subset of nodes to apply the function to
    * @param[out] result the results
    */
-  template< typename LEAF >
+  template< typename LEAF, typename POLICY = serialPolicy >
   void evaluateT( dataRepository::Group const & group,
                   real64 const time,
                   SortedArrayView< localIndex const > const & set,
-                  real64_array & result ) const;
+                  arrayView1d< real64 > const & result ) const;
 
   virtual void postProcessInput() override { initializeFunction(); }
 
 };
 
-template< typename LEAF >
+template< typename LEAF, typename POLICY >
 void FunctionBase::evaluateT( dataRepository::Group const & group,
                               real64 const time,
                               SortedArrayView< localIndex const > const & set,
-                              real64_array & result ) const
+                              arrayView1d< real64 > const & result ) const
 {
-  real64 const * input_ptrs[4];
-  localIndex varSize[4] = {0, 0, 0, 0};
-  int timeVar[4] = {1, 1, 1, 1};
+  real64 const * inputPtrs[MAX_VARS]{};
+  localIndex varSize[MAX_VARS]{};
+  localIndex varStride[MAX_VARS][2]{};
 
-  arrayView1d< string const > const & inputVarNames = this->getReference< string_array >( dataRepository::keys::inputVarNames );
-  localIndex const numVars = LvArray::integerConversion< localIndex >( inputVarNames.size());
-  localIndex groupSize = group.size();
+  integer const numVars = LvArray::integerConversion< integer >( m_inputVarNames.size() );
```

Comment:
This is a longstanding issue with the way we use the same `Array` type alias for everything, including XML inputs and model parameters that do not scale with mesh size. I think we could define a new alias for these, using `integer` as `INDEX_TYPE` (and so `size()` would return an `integer` automatically),

(There's also a related issue that some of these arrays don't need the multiple-memory-space semantics, in particular any string inputs which are unusable on device anyway. We could start using `MallocBuffer` instead of `ChaiBuffer` in the associated array types, but I'm not sure this has any practical benefits.)